lib/cli/template/_LOCAL/wao-bind.js

```diff
@@ -0,0 +1,286 @@
+        } else if (retVar.length == 0) {
+          throw "Bind data is not found.";
+        } else {
+          throw "Bind data is too many.";
+        }
+      }
+      return retVar;
+    }
+
+    // -------------------------------------
+    // data-wao-iteratorの実行
+    // -------------------------------------
+    var iterate = function($this, iteratorPath) {
+      // 予備動作
+      var $parent = $this.parent(); // 親要素を取得する
+      var $template = $this.removeAttr('data-wao-iterator').clone(); // 自分自身をテンプレート化
+      var $stack = $('<div />');
+      var iterator = getIterator(iteratorPath); // 繰り返し用の変数を取得しておく
+      $this.nextAll().each(function() {
+        // TODO kimura 削除対象にwao属性が残っていたらエラーにする予定
+        if ($(this).attr('data-wao') != 'keep' && !$(this).attr(
+            'data-wao-bind')) {
+          $(this).remove();
+        }
+      });
+      // 繰り返し処理本体
+      // iteratorは配列であることが保障されているのでfor-inを使用
+      if (iterator) {
+        for (var idx in iterator) {
+          var $temporary = $template.clone(); // テンプレートをコピー
+          // bind処理用にiterator情報を作成する
+          var mysettings = $.extend(settings, {});
+          mysettings.iterators[iteratorPath] = {
+            'this': iterator, // iterator変数（配列）
+            'idx': idx // インデクス値
+          };
+          var tmptag = mysettings.tagStack.pop();
+          $temporary.wao_bind(mysettings); // 繰り返しbind処理
+          mysettings.tagStack.push(tmptag);
+          $stack.append($temporary); // 1件分の生成結果を親要素に追加
+        }
+        $this.after($stack.html()).remove();
+      } else {
+        $this.remove();
+      }
+    }
+
+    // -------------------------------------
+    // iteratorを取得する
+    // -------------------------------------
+    var getIterator = function(iteratorPath) {
+      var iterator = getVar(iteratorPath);
+      // iteratorとして使用する変数は配列でなければならない
+      if (iterator === undefined) {
+        return iterator;
+      }
+      return ($.isArray(iterator)) ? iterator : [iterator];
+    }
+
+    // -------------------------------------
+    // bind用の編集を返却する
+    // -------------------------------------
+    var bindVar = function(variable) {
+      // bind対象の変数が配列の場合には変換が必要
+      if ($.isArray(variable)) {
+        if (variable.length == 1) { // 要素数が1個ならbind可能
+          variable = variable[0];
+        } else
+        // 要素数が0または2以上の配列をbindしようとしたらエラー
+        if (variable.length == 0) {
+          throw "Bind data is not found.";
+        } else {
+          throw "Bind data is too many.";
+        }
+      }
+      return $('<div>').text(variable).html();
+    }
+
+    /**
+     * クエリストリングのバインド処理
+     */
+    var bindGetParam = function(target) {
+      // ?hoge=hoge&hoge=hoge....部分を抜き出す
+      var getparam = target.match(/\?([^=]+=([^&]*)?&?)*/);
+      var getparamR = '';
+      if (getparam) { // URLにGETパラメタが含まれていれば
+        getparamR = '?';
+        getparam = getparam[0].replace(/^\?/, '').split('&'); // hoge=hogeに分割して配列化
+        for (var i in getparam) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.